aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -2863,6 +2874,19 @@ void svd_kernel(const Tensor& A,
       Vh._set_conj(!Vh.is_conj());
     }
   }
+#elif defined(USE_HIPSOLVER)
+  bool use_magma = at::globalContext().linalgPreferredBackend() == at::LinalgBackend::Magma;
+  use_magma = use_magma || (A.size(-1) >= 50 && A.size(-2) >= 50);
```

Comment:
Let's mention in a comment the reason why we have the additional condition on `A.size()` here. Also, it seems like we could have avoided the duplication of the entire code by doing something like:
```
#if defined(USE_CUSOLVER) || defined(USE_HIPSOLVER)
  // We always use cuSOLVER unless the user has specified they want to use MAGMA
  if (at::globalContext().linalgPreferredBackend() == at::LinalgBackend::Magma
#ifdef USE_HIPSOLVER
  // COMMENT
  || (A.size(-1) >= 50 && A.size(-2) >= 50)
#endif
  ) {
    svd_magma(A, full_matrices, compute_uv, U, S, Vh, info);
```